File: id-card-toolkit-ios-sdk-v3.1.6/id-card-toolkit-ios-sdk-v3.1.6/samples/IDCardToolkitObjCApp/IDCardToolkitObjCApp/ViewController/ConfigCertExpiryDateViewController.m
//
//  ConfigCertExpiryDateViewController.m
//  IDCardToolkitObjCApp
//
//  Created by Federal Authority For Identity and Citizenship on 24/07/19.
//  Copyright © 2019 Federal Authority For Identity and Citizenship. All rights reserved.
//

#import "ConfigCertExpiryDateViewController.h"

@interface ConfigCertExpiryDateViewController ()
@property(nonatomic,strong)Utils * utils;
@end

@implementation ConfigCertExpiryDateViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.title=@"Config Certificate Expiry Date";
    self.navigationItem.hidesBackButton = YES;
    self.navigationItem.leftBarButtonItem = [[UIBarButtonItem alloc] initWithImage:[[UIImage imageNamed:@"backbutton"] imageWithRenderingMode:UIImageRenderingModeAlwaysOriginal]
                                                                             style:UIBarButtonItemStylePlain
                                                                            target:self
                                                                            action:@selector(goBack)];
    self.getExpiryDateButton.layer.cornerRadius=22.0;
    [self.getExpiryDateButton.layer setShadowOffset:CGSizeMake(5, 5)];
    [self.getExpiryDateButton.layer setShadowColor:[[UIColor lightGrayColor] CGColor]];
    [self.getExpiryDateButton.layer setShadowOpacity:0.5];
   // self.getExpiryDateButton.layer.borderWidth=1.0;
    
    AppDelegate *aDelegate = (AppDelegate *)[UIApplication sharedApplication].delegate;
    self.model = aDelegate.model;
    self.utils=aDelegate.utils;
    // Do any additional setup after loading the view.
}
-(void)viewWillAppear:(BOOL)animated {
    
    [super viewWillAppear:YES];
    self.subView.hidden=YES;
}
- (IBAction)getExpiryDatebuttonAction:(id)sender {
    
    [self.utils ShowProgressBar:LOADINGDATA andView:self.view];

    dispatch_async(dispatch_get_global_queue( DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^(void) {
        @try {
            Toolkit *toolkit = self.model.toolkitShared;
            [toolkit getConfigCertificateExpiryDate];
            
            dispatch_async(dispatch_get_main_queue(), ^{
                self.vgCertExpDate.text=[toolkit getConfig_vg_cert_expiry];
                self.lvCertExpdate.text=[toolkit getConfig_lv_cert_expiry];
                self.tlsCertExpdate.text=[toolkit getServer_tls_cert_expiry];
                self.agCertExpDate.text=[toolkit getConfig_ag_cert_expiry];
                self.licenseExpdate.text=[toolkit getLicense_expiry];
                self.subView.hidden=NO;
            });// update UI main queue
        }//try
        @catch (NSException *exception) {
            NSString *exe =[NSString stringWithFormat:@"%@\n%@\n%@",exception.name,exception.description,exception.userInfo];
            [AlertView showAlertTitle:ALERT withMessage:exe onView:self];
        }// catch
        @finally {
            dispatch_async(dispatch_get_main_queue(), ^{
                [self.utils DismissProgressBar];
            });// update UI main queue
        }// finally
    });// background queue
}
- (void)goBack {
    [self.navigationController popViewControllerAnimated:YES];
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: id-card-toolkit-ios-sdk-v3.1.6/id-card-toolkit-ios-sdk-v3.1.6/samples/ICASwiftApp/ICASwiftApp/ConfigCertExpiryDateViewController.swift
//
//  ConfigCertExpiryDateViewController.swift
//  ICASwiftApp
//
//  Created by Digital Trust Tech on 10/05/22.
//

import UIKit

class ConfigCertExpiryDateViewController: UIViewController {
    
    @IBOutlet weak var vgCertExpDate: UILabel!
    @IBOutlet weak var lvCertExpDate: UILabel!
    @IBOutlet weak var tlsCertExpDate: UILabel!
    @IBOutlet weak var agCertExpDate: UILabel!
    @IBOutlet weak var licenseExpDate: UILabel!
    @IBOutlet weak var getExpiryDateButton: UIButton!
    @IBOutlet weak var subView: UIStackView!
    @IBOutlet weak var view_Roundedcorners: UIView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.view_Roundedcorners.cornered_View([.layerMaxXMaxYCorner, .layerMinXMaxYCorner], radius: 20, borderColor: .red, borderWidth: 0)
        
        getExpiryDateButton.layer.cornerRadius = 22.0
        getExpiryDateButton.layer.shadowOffset = CGSize(width: 5.0, height: 5.0)
        getExpiryDateButton.layer.shadowColor = UIColor.lightGray.cgColor
        getExpiryDateButton.layer.shadowOpacity = 0.5

        // Do any additional setup after loading the view.
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(true)
        self.subView.isHidden = true
    }
    
    @IBAction func getExpiryDateButtonAction(_ sender: Any) {
        
        if toolkit == nil  {
            self.view.makeToast(Constants.toolkitNotInitilized)
            return
        }//if
        
        utils.showProgressBar(Constants.LoadingData, andView: self.view)
        DispatchQueue.global(qos: .background).async {
            do {
                try toolkit?.getConfigCertificateExpiryDate()
                DispatchQueue.main.async {
                    self.vgCertExpDate.text = toolkit?.getConfig_vg_cert_expiry()
                    self.lvCertExpDate.text = toolkit?.getConfig_lv_cert_expiry()
                    self.tlsCertExpDate.text = toolkit?.getServer_tls_cert_expiry()
                    self.agCertExpDate.text = toolkit?.getConfig_ag_cert_expiry()
                    self.licenseExpDate.text = toolkit?.getLicense_expiry()
                    self.subView.isHidden=false;
                    utils.dismissProgressBar()
                }// update UI main queue
            }//do
            catch let error as NSError{
                DispatchQueue.main.async {
                    let err="\(error.domain)\n\(error.code)\n\(error.userInfo)"
                    AlertView.showAlertTitle(Constants.Alert, withMessage:err,onView: self)
                    utils.dismissProgressBar()
                }// update UI main queue
            }//catch
        }//background queue
    }
    
    @IBAction func btn_BackAction(_ sender: Any) {
        navigationController?.popViewController(animated: true)
    }
}
